streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamPeek.java

```diff
@@ -35,10 +39,35 @@ public KStreamPeek(final ForeachAction<K, V> action) {
     }
 
     private class KStreamPeekProcessor extends AbstractProcessor<K, V> {
+        
+        @Override
+        public void init(ProcessorContext context) {
+            super.init(context);
+            if (action instanceof PrintForeachAction) {
+                ((PrintForeachAction) action).setContext(context);
```

Comment:
Sorry. Typo. I meant the "cast" not the "case"... Right now, after `instanceof ` you cast each time you access `action`. Would be more readable like
```
if (action instanceof PrintForeachAction) {
    final PrintForeachAction printAction = (PrintForeachAction) action;
    printAction.setContext(context);
    ...
}
```